src/sif/autocost.cc

```diff
@@ -454,7 +456,9 @@ bool AutoCost::AllowedReverse(const baldr::DirectedEdge* edge,
       opp_edge->surface() == Surface::kImpassable || IsUserAvoidEdge(opp_edgeid) ||
       (!allow_destination_only_ && !pred.destonly() && opp_edge->destonly()) ||
       (pred.closure_pruning() && IsClosed(opp_edge, tile)) ||
-      (exclude_unpaved_ && !pred.unpaved() && opp_edge->unpaved()) || !IsHOVAllowed(opp_edge)) {
+      (exclude_unpaved_ && !pred.unpaved() && opp_edge->unpaved()) || !IsHOVAllowed(opp_edge) ||
+      (has_excludes_ && CheckExclusions(opp_edge, pred, exclude_bridges_, exclude_tunnels_,
```

Comment:
since `has_excludes_` is accessible from inside `CheckExclusions` can we just put it in there? also why do we need to pass these flags to it, its a member function, it has access to all of them, the only operand should be `opp_edge` right?

Reply:
I dont know why these flags are implemented as parameters to the function, I think it was an oversight, I took them out and only left edge and pred in it.
I now put `has_excludes_` in the `CheckExclusions`-Function and I also went ahead and removed the `IsExcludedRoadClass`-Function and inlined it in `CheckExclusions` so it is all implemented in there.